app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1302,12 +1309,32 @@ private Cursor makeStreamCursor(CharSequence streamName)
                 .getDao(Stream.class)
                 .queryRaw(
                         "SELECT rowid _id, * FROM streams WHERE "
+                                + Stream.NAME_FIELD
+                                + " LIKE ? ESCAPE '\\' ORDER BY "
+                                + Stream.NAME_FIELD + " COLLATE NOCASE",
+                        DatabaseHelper.likeEscape(streamName.toString()) + "%")
+                .closeableIterator().getRawResults()).getRawCursor();
+    }
+
+
+    private boolean isSubscribed(CharSequence streamName)
+            throws SQLException {
+        if (streamName == null) {
+            streamName = "";
+        }
+
+        Cursor cursor =  ((AndroidDatabaseResults) app
+                .getDao(Stream.class)
+                .queryRaw(
+                        "SELECT rowid _id, * FROM streams WHERE "
                                 + Stream.SUBSCRIBED_FIELD + " = 1 AND "
                                 + Stream.NAME_FIELD
                                 + " LIKE ? ESCAPE '\\' ORDER BY "
```

Comment:
Use `where Stream.NAME_FIELD =` instead of `like`
Suppose if a user is subscribed to `test here` and not `test` stream, now when you enter "test" in stream EditText you will be able to send the message.